src/utils/galileo-logger.ts

```diff
@@ -20,18 +20,130 @@ class GalileoLoggerConfig {
   public experimentId?: string;
 }
 
+/**
+ * Higher-order function that wraps a method to skip execution if logging is disabled
+ * @param fn The original method
+ * @param defaultValueFn A function that returns the default value when logging is disabled
+ */
+function skipIfDisabled<T, Args extends any[]>(
+  fn: (this: GalileoLogger, ...args: Args) => T,
+  defaultValueFn: (args: Args) => T
+): (this: GalileoLogger, ...args: Args) => T {
+  return function (this: GalileoLogger, ...args: Args): T {
+    if (this.isLoggingDisabled()) {
+      console.warn('Logging is disabled, skipping execution of', fn.name);
+      return defaultValueFn(args);
+    }
+    return fn.apply(this, args);
+  };
+}
+
+/**
+ * Higher-order function that wraps an async method to skip execution if logging is disabled
+ * @param fn The original async method
+ * @param defaultValueFn A function that returns the default value when logging is disabled
+ */
+function skipIfDisabledAsync<T, Args extends any[]>(
+  fn: (this: GalileoLogger, ...args: Args) => Promise<T>,
+  defaultValueFn: (args: Args) => T
+): (this: GalileoLogger, ...args: Args) => Promise<T> {
+  return async function (this: GalileoLogger, ...args: Args): Promise<T> {
+    if (this.isLoggingDisabled()) {
+      console.warn('Logging is disabled, skipping execution of', fn.name);
+      return defaultValueFn(args);
+    }
+    return await fn.apply(this, args);
+  };
+}
+
 class GalileoLogger {
   private projectName?: string;
   private logStreamName?: string;
   private experimentId?: string;
   private client = new GalileoApiClient();
   private parentStack: StepWithChildSpans[] = [];
   public traces: Trace[] = [];
+  private loggingDisabled: boolean;
 
   constructor(config: GalileoLoggerConfig = {}) {
     this.projectName = config.projectName;
     this.logStreamName = config.logStreamName;
     this.experimentId = config.experimentId;
+    try {
+      // Logging is disabled if GALILEO_DISABLE_LOGGING is defined, is not an empty string, and not set to '0' or 'false'
+      const disableLoggingValue =
+        typeof process !== 'undefined' && typeof process.env !== 'undefined'
+          ? process.env.GALILEO_DISABLE_LOGGING
+            ? process.env.GALILEO_DISABLE_LOGGING.trim()
+            : undefined
+          : undefined;
+
+      this.loggingDisabled =
```

Comment:
And continuing with the unnecessary suggestions:

```suggestion
       this.loggingDisabled = !['', '0', 'false'].includes(disableLoggingValue?.toLowerCase() ?? '');
```

Feel free to discard them.